### PR TITLE
Improve test coverage of sync.py

### LINKED ISSUE #1913: Improve test coverage of `sync.py`
### Description of feature

Relates to #1790, #1697

### PR DESCRIPTION
**Pull request by Aratz**
This PR improves the code coverage of the `sync` module and fixes a couple of bugs along the way.

Only the `sync` function remains untested but this one is kind of a beast so I leave it for now.

Closes #1913 

## PR checklist

- [X] This comment contains a description of changes (with reason)
- [X] `CHANGELOG.md` is updated
- [X] If you've fixed a bug or added code that should be tested, add tests!

### CODE REVIEW

**Review by mirpedrol — APPROVED** [review]
LGTM :)

File: tests/test_sync.py [hunk 1]
```diff
@@ -153,11 +161,55 @@ def test_push_template_branch_error(self):
...
+        """Try creating a merge base branch"""
+        psync = nf_core.sync.PipelineSync(self.pipeline_dir)
+        psync.inspect_sync_dir()
+        psync.get_wf_config()
+
+        psync.create_merge_base_branch()
+
+        assert psync.merge_branch in psync.repo.branches
+
+    def test_create_merge_base_branch_thrice(self):
```
Comment (mashehu) [inline tests/test_sync.py:178]: Just curious, why would it work once, but not three times?

**Review by mashehu — APPROVED** [review]

File: tests/test_sync.py (hunk 1 above)
Comment (Aratz) [reply tests/test_sync.py:178]: The first time you call this function, the merge branch does not exist yet (it is only created at the end of this function), so you don't go into the `if`-statement at L274. Also, the second time you call this function, the existing merge branch only has the base format, i.e. without the `-{branch_no}` at the end, so you need to call it a third time to make sure this is picked up.

https://github.com/Aratz/nf-core-tools/blob/dev/nf_core/sync.py#L274-L286

Alternatively, we could set the merge branch format to include the branch number suffix from the beginning, so you only need to test it twice to cover all the code.

Comment (mashehu) [reply tests/test_sync.py:178]: makes totally sense now. thanks! Do you mind putting this in as a comment, so we don't forget about the logic. also there is a merge conflict now in the changelog. but than we are good to go!

Comment (Aratz) [reply tests/test_sync.py:178]: Yes will do! :+1:

Comment (Aratz) [reply tests/test_sync.py:178]: Should be all good now

**Review by mashehu — APPROVED** [review]

**Pull request merged**